backup/Command1.java

```diff
@@ -0,0 +1,23 @@
+public class Command {
+    protected CommandType type;
+    protected int index;
+    protected Task task;
+
+    public Command(CommandType type, Task task) {
+        this.type = type;
+        this.index = -1;
+        this.task = task;
+    }
+
+    public Command(CommandType type, int index) {
+        this.type = type;
+        this.index = index;
+        this.task = null;
+    }
+
+    public Command(CommandType type) {
```

Comment:
Good way of using constructor overloading to handle different kinds of conditions rather than throwing exceptions everywhere.